new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CleanupUtil.java

```diff
@@ -0,0 +1,524 @@
+
+package oracle.weblogic.kubernetes.utils;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1Ingress;
+import io.kubernetes.client.openapi.models.V1ConfigMap;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1Job;
+import io.kubernetes.client.openapi.models.V1PersistentVolume;
+import io.kubernetes.client.openapi.models.V1PersistentVolumeClaim;
+import io.kubernetes.client.openapi.models.V1ReplicaSet;
+import io.kubernetes.client.openapi.models.V1Role;
+import io.kubernetes.client.openapi.models.V1RoleBinding;
+import io.kubernetes.client.openapi.models.V1Secret;
+import io.kubernetes.client.openapi.models.V1Service;
+import io.kubernetes.client.openapi.models.V1ServiceAccount;
+import oracle.weblogic.domain.Domain;
+import oracle.weblogic.kubernetes.TestConstants;
+import oracle.weblogic.kubernetes.actions.TestActions;
+import oracle.weblogic.kubernetes.actions.impl.primitive.HelmParams;
+import oracle.weblogic.kubernetes.actions.impl.primitive.Kubernetes;
+import org.awaitility.core.ConditionFactory;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+import static org.awaitility.Awaitility.with;
+
+/**
+ * CleanupUtil is used for cleaning up all the Kubernetes artifacts left behind by the integration tests.
+ *
+ */
+public class CleanupUtil {
+
+  /**
+   * Cleanup all artifacts in the Kubernetes cluster.
+   *
+   * <p>Tries to gracefully delete the WebLogic domains and WebLogic Operator in the namespaces, if found.
+   * Then deletes everything in the namespaces.
+   *
+   * <p>Waits for the deletion task to be completed until up to 3 minutes.
+   *
+   * @param namespaces list of namespaces
+   */
+  public static void cleanup(List<String> namespaces) {
+    try {
+      // If namespace list is empty or null return
+      if (namespaces == null || namespaces.isEmpty()) {
+        return;
+      }
+      // iterate through the namespaces and delete domain as a
+      // first entity if its not operator namespace
+      for (var namespace : namespaces) {
+        if (!isOperatorNamespace(namespace)) {
+          deleteDomains(namespace);
+        }
+      }
+      // iterate through the namespaces and delete operator if
+      // its operator namespace.
+      for (var namespace : namespaces) {
+        if (isOperatorNamespace(namespace)) {
+          uninstallOperator(namespace);
+        }
+      }
+
+      // Delete artifacts in namespace used by the test class
+      for (var namespace : namespaces) {
+        deleteNamespacedArtifacts(namespace);
+      }
+
+      Thread.sleep(30 * 1000); // Using Thread.sleep for a one time 30 sec sleep.
```

Comment:
```
try { 
  Thread.sleep(30 *1000);
} catch (InterruptedException e) { 
  //ignore 
}
```

Reply:
also, why? :)

Reply:
Otherwise I need to put this wait in the pollDelay which will make it wait for every namespace. Since the delete on the namespaces was already called a one time wait is good enough.
 
ConditionFactory withStandardRetryPolicy = with().pollDelay(0, SECONDS)
          .and().with().pollInterval(10, SECONDS)
          .atMost(10, MINUTES).await();